RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1236,6 +1242,8 @@ class DeepTauId : public deep_tau::DeepTauBase {
     edm::Handle<double> rho;
     event.getByToken(rho_token_, rho);
 
+    unsigned int eventnr = static_cast<unsigned int>(event.id().event());
```

Comment:
I think that you do not need cast here - the number is going to be used just for printout so you can use here and in the following code const reference here `auto const&` and below in declaration of the method an appropriate type:
The `unsigned int` type can be too short event number - see  DataFormats/Provenance/interface/RunLumiEventNumber.h. Please use `edm::EventNumber_t` (which is `unsigned long long`).